packages/cfd/src/Components/__tests__/cfd-personal-details-form.spec.js

```diff
@@ -0,0 +1,250 @@
+        expect(screen.queryByTestId('dt_cfd_details_form_description')).not.toBeInTheDocument();
+    });
+
+    it('should disable Citizenship and Tax residence fields if they were submitted earlier & is_fully_authenticated is true', () => {
+        const values = {
+            citizen: 'Indonesia',
+            tax_residence: 'Indonesia',
+            tax_identification_number: '',
+            account_opening_reason: '',
+        };
+        render(<CFDPersonalDetailsForm {...props} is_fully_authenticated value={values} />);
+
+        expect(screen.getByRole('textbox', { name: /citizenship/i })).toBeDisabled();
+        expect(screen.getByRole('textbox', { name: /tax residence/i })).toBeDisabled();
+        expect(screen.getByRole('textbox', { name: /tax identification number/i })).toBeEnabled();
+        expect(screen.getByText(/account opening reason/i)).toBeEnabled();
+        expect(screen.getByRole('button', { name: /next/i })).toBeEnabled();
+    });
+
+    it('should show an error message received from server that is passed via props as form_error', () => {
+        const form_error = 'Form submission failed.';
+        render(<CFDPersonalDetailsForm {...props} form_error={form_error} />);
+
+        expect(screen.getByTestId('form_submit_error')).toHaveClass('dc-icon');
+        expect(screen.getByText(form_error)).toBeInTheDocument();
+        expect(screen.getByRole('button', { name: /next/i })).toBeEnabled();
+    });
+
+    it('should enable the Next button for form submission when all required fields are filled', async () => {
+        render(<CFDPersonalDetailsForm {...props} />);
+
+        const citizenship_input = screen.getByRole('textbox', { name: /citizenship/i });
+        const tax_residence_input = screen.getByRole('textbox', { name: /tax residence/i });
+        const tax_id_input = screen.getByRole('textbox', { name: /tax identification number/i });
+        const opening_reason_input = screen.getByText(/account opening reason/i);
+        const next_button = screen.getByRole('button', { name: /next/i });
+
+        fireEvent.change(citizenship_input, { target: { value: 'Indonesia' } });
+        fireEvent.click(next_button);
+
+        expect(await screen.findByText(tax_residence_required_error)).toBeInTheDocument();
+        expect(await screen.findByText(tax_id_required_error)).toBeInTheDocument();
+        expect(await screen.findByText(opening_reason_required_error)).toBeInTheDocument();
+        expect(screen.queryByText(citizenship_required_error)).not.toBeInTheDocument();
+        expect(next_button).toBeDisabled();
+
+        fireEvent.change(tax_residence_input, { target: { value: 'Indonesia' } });
+        fireEvent.click(next_button);
+
+        expect(await screen.findByText(tax_id_required_error)).toBeInTheDocument();
+        expect(await screen.findByText(opening_reason_required_error)).toBeInTheDocument();
+        expect(screen.queryByText(citizenship_required_error)).not.toBeInTheDocument();
```

Comment:
same here

Reply:
when the Next button is clicked, onBlur event happens. that's why the test is passing